### PR TITLE
Fix #2 - expose route

### LINKED ISSUE #2: Provide some basic routes
I think this module should provide some default flows.

### PR DESCRIPTION
**Pull request by allevo**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,14 +2,72 @@
...
+    const state = request.query.state
+    const fastify = this
+
+    checkStateFunction(state, function (err) {
+      if (err) {
+        callback(err)
+        return
+      }
+      cbk(fastify[name], code, callback)
+    })
```
Comment (mcollina) [inline index.js]: I would love if this supported Promises as well.

Comment (allevo) [reply index.js]: Done

File: index.js [hunk 2]
```diff
@@ -18,6 +104,17 @@ const oauthPlugin = fp(function (fastify, options, next) {
...
+function promisify (func) {
+  return function (arg1) {
+    return new Promise(function (resolve, reject) {
+      func(arg1, function (err, token) {
+        if (err) reject(err)
+        else resolve(token)
+      })
+    })
+  }
+}
```
Comment (mcollina) [inline index.js]: You can use https://www.npmjs.com/package/util.promisify instead.

File: test.js [hunk 3]
```diff
@@ -8,89 +8,294 @@ const oauthPlugin = require('./index')
...
+
+    t.tearDown(fastify.close.bind(fastify))
+
+    makeRequests(t, fastify)
   })
 
-  t.test('ok', t => {
-    t.plan(4)
+  t.test('promise', t => {
+    const fastify = createFastify({ logger: { level: 'silent' } })
```
Comment (mcollina) [inline test.js]: there is not need to pass level: silent, it's the default.

File: index.js (hunk 2 above)
Comment (allevo) [reply index.js]: done. es6-promisify used

**Pull request merged**